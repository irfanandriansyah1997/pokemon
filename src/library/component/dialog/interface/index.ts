import { FC } from 'react';

import { IEventComponent } from '@/library/interface/general';

export type IEventClose = IEventComponent<'on-close'>;

/**
 * Dialog Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.08.01
 */
export interface IDialogProps {
  on(param: IEventClose): void;
  show: boolean;
  title: string;
}

/**
 * Dialog Export Default Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.08.01
 */
export type IDialogExportDefault = FC<IDialogProps> & {
  Body: FC;
  Footer: FC;
};
