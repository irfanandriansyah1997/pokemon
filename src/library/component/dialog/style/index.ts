import styled from '@emotion/styled';

import { COLOR } from '@/library/styles/constant';

export const DialogWrapper = styled.div<{ show: boolean }>(({ show }) => ({
  '&::before': {
    content: `''`,
    display: `inline-block`,
    height: `100%`,
    verticalAlign: `middle`,
    width: 0
  },
  bottom: 0,
  left: 0,
  opacity: show ? 1 : 0,
  outline: 0,
  overflow: `auto`,
  position: `fixed`,
  right: 0,
  textAlign: `center`,
  top: 0,
  transform: show ? `translate(0)` : `translateY(100%)`,
  transition: `all 0.6s`,
  visibility: show ? `visible` : `hidden`,
  zIndex: 99999999
}));

export const DialogStyled = styled.div<{ show: boolean }>(({ show }) => ({
  background: `#fff`,
  borderRadius: 8,
  boxShadow: `0 -2px 16px 0 rgba(0, 0, 0, 0.1)`,
  display: `inline-block`,
  flex: 1,
  margin: `32px auto`,
  maxWidth: 300,
  opacity: show ? 1 : 0,
  overflow: `hidden`,
  position: `relative`,
  textAlign: `left`,
  top: 0,
  transition: `transform 200ms ease-in-out 0s, opacity 200ms ease-in-out 0s`,
  verticalAlign: `middle`,
  width: `100%`
}));

export const DialogHeader = styled.div({
  alignItems: `center`,
  background: COLOR.white,
  borderBottom: `1px dashed #ddd`,
  display: `flex`,
  justifyContent: `space-between`,
  padding: 16,
  position: `relative`,
  zIndex: 9
});

export const DialogClose = styled.div({
  alignItems: `center`,
  display: `flex`,
  justifyContent: `center`,
  position: `absolute`,
  right: 16,
  top: `50%`,
  transform: `translateY(-50%)`
});

export const DialogBody = styled.div({
  padding: 16,
  wordBreak: `break-word`
});

export const DialogFooter = styled.div({
  display: `flex`,
  justifyContent: `flex-end`,
  padding: `12px 16px`
});
