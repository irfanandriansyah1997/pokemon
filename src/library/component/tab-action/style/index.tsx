import styled from '@emotion/styled';

import {
  ITabActionItemProps,
  ITabActionWrapperProps
} from '@/library/component/tab-action/interface';
import { COLOR } from '@/library/styles/constant';

export const TabActionWrapper = styled.div<ITabActionWrapperProps>(
  ({ count }) => ({
    alignItems: `center`,
    backgroundColor: COLOR.white,
    borderBottom: `1px solid #ddd`,
    boxShadow: `0 0px 1px rgba(40, 41, 61, 0.08), 0 2px 2px rgba(96, 97, 112, 0.08)`,
    display: `grid`,
    gap: 12,
    gridTemplateColumns: `repeat(${count}, 1fr)`,
    height: 40,
    rowGap: 0,
    width: `100%`
  })
);

export const TabActionItem = styled.div<ITabActionItemProps>(({ active }) => ({
  alignItems: `center`,
  borderBottom: `2px solid ${active ? COLOR.red : `transparent`}`,
  display: `flex`,
  height: `100%`,
  justifyContent: `center`
}));
