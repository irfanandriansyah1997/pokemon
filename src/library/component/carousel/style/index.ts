import styled from '@emotion/styled';

import Next from '@/assets/images/next.svg';
import Prev from '@/assets/images/prev.svg';
import { ICarouselWrapper } from '@/library/component/carousel/interface';
import { COLOR } from '@/library/styles/constant';

export const CarouselWrapper = styled.div<ICarouselWrapper>(
  ({ showCarousel }) => ({
    '> div:first-of-type': {
      alignItems: `center`,
      display: `flex`,
      transform: `translateX(0%)`,
      transition: `0.3s cubic-bezier(0.7, 0.3, 0, 1)`,
      width: `100%`
    },
    alignItems: `center`,
    display: `flex`,
    height: 200,
    opacity: showCarousel ? 1 : 0,
    position: `relative`,
    transition: `0.3s cubic-bezier(0.7, 0.3, 0, 1)`,
    width: `100%`
  })
);

export const CarouselItem = styled.div({
  '&::before': {
    background: `rgba(0, 0, 0, 0.1)`,
    borderRadius: `50%`,
    content: `''`,
    height: 160,
    position: `absolute`,
    width: 160
  },
  alignItems: `center`,
  display: `flex`,
  flex: `0 0 100%`,
  img: {
    width: 130
  },
  justifyContent: `center`,
  position: `relative`
});

export const CarouselPrev = styled.div<{ show: boolean }>(({ show }) => ({
  '&::before': {
    background: `url(${Prev}) no-repeat`,
    backgroundSize: `10px 10px`,
    content: `''`,
    height: 10,
    left: `50%`,
    position: `absolute`,
    top: `50%`,
    transform: `translate(-50%, -50%)`,
    width: 10
  },
  alignItems: `center`,
  background: COLOR.white,
  borderRadius: `50%`,
  boxShadow: `0 2px 8px rgba(40, 41, 61, 0.08), 0 16px 24px rgba(96, 97, 112, 0.32)`,
  display: `flex`,
  flex: `0 0 100%`,
  height: 45,
  justifyContent: `center`,
  left: 10,
  opacity: show ? 1 : 0,
  position: `absolute`,
  top: `60%`,
  transform: `translateY(-50%)`,
  transition: `0.3s cubic-bezier(0.7, 0.3, 0, 1)`,
  width: 45
}));

export const CarouselNext = styled.div<{ show: boolean }>(({ show }) => ({
  '&::before': {
    background: `url(${Next}) no-repeat`,
    backgroundSize: `10px 10px`,
    content: `''`,
    height: 10,
    left: `50%`,
    position: `absolute`,
    top: `50%`,
    transform: `translate(-50%, -50%)`,
    width: 10
  },
  alignItems: `center`,
  background: COLOR.white,
  borderRadius: `50%`,
  boxShadow: `0 2px 8px rgba(40, 41, 61, 0.08), 0 16px 24px rgba(96, 97, 112, 0.32)`,
  display: `flex`,
  flex: `0 0 100%`,
  height: 45,
  justifyContent: `center`,
  opacity: show ? 1 : 0,
  position: `absolute`,
  right: 10,
  top: `60%`,
  transform: `translate(-50%, -50%)`,
  transition: `0.3s cubic-bezier(0.7, 0.3, 0, 1)`,
  width: 45
}));

export const CarouselIndicator = styled.div<{ indicator: number }>(
  ({ indicator }) => ({
    '> div:first-of-type': {
      background: indicator === 0 ? COLOR.white : undefined
    },
    '> div:last-of-type': {
      background: indicator === 1 ? COLOR.white : undefined
    },
    alignItems: `center`,
    bottom: -20,
    display: `flex`,
    div: {
      border: `1px solid ${COLOR.white}`,
      borderRadius: `50%`,
      margin: `0 5px`,
      minHeight: 10,
      minWidth: 10,
      transition: `0.3s cubic-bezier(0.7, 0.3, 0, 1)`
    },
    justifyContent: `center`,
    left: `50%`,
    position: `absolute`,
    transform: `translateX(-50%)`,
    width: 50
  })
);
