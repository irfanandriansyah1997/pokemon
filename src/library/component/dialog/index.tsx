import { useEffect } from 'react';

import { Text } from '@/library/styles/general.styles';
import { PokemonBackdropStyle } from '@/library/styles/pokemon.styles';

import { IDialogExportDefault } from './interface';
import {
  DialogBody,
  DialogClose,
  DialogFooter,
  DialogHeader,
  DialogStyled,
  DialogWrapper
} from './style';

/**
 * Dialog Component
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.08.01
 */
const Dialog: IDialogExportDefault = ({ children, on, show, title }) => {
  /**
   * Simulate Close
   * @returns {void}
   * */
  const onClose = () => {
    if (show)
      on({
        event: `on-close`
      });
  };

  useEffect(() => {
    if (show && document) {
      document.body.style.overflow = `hidden`;
    } else {
      document.body.style.overflow = `inherit`;
    }
  }, [show]);

  return (
    <>
      <PokemonBackdropStyle
        show={show}
        color="rgba(0,0,0,0.25)"
        zIndex={99999999}
        onClick={onClose}
      />
      <DialogWrapper show={show}>
        <DialogStyled show={show}>
          <DialogHeader>
            <Text fontSize="normal" color="primary" fontWeight={500}>
              {title}
            </Text>
            <DialogClose onClick={onClose}>
              <Text fontSize="medium">&times;</Text>
            </DialogClose>
          </DialogHeader>
          {children}
        </DialogStyled>
      </DialogWrapper>
    </>
  );
};

Dialog.Footer = DialogFooter;
Dialog.Body = DialogBody;

export default Dialog;
