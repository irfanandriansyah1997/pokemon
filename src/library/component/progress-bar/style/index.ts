import styled from '@emotion/styled';

import { IProgressBarProps } from '@/library/component/progress-bar/interface';
import { COLOR } from '@/library/styles/constant';

export const ProgressBarWrapper = styled.div({
  backgroundColor: COLOR.light,
  borderRadius: 5,
  boxShadow: `0 0 1px rgba(40, 41, 61, 0.08), 0 0.5px 2px rgba(96, 97, 112, 0.16)`,
  height: 4,
  overflow: `hidden`,
  width: `100%`
});

export const ProgressBarContent = styled.div<IProgressBarProps>(
  ({ maxValue, value }) => {
    const percentage = (value / maxValue) * 100;
    let color = ``;

    if (percentage <= 50) {
      color = `#fb8500`;
    } else if (percentage <= 75) {
      color = `#ffb703`;
    } else {
      color = `#219ebc`;
    }

    return {
      backgroundColor: color,
      borderRadius: 5,
      height: 4,
      overflow: `hidden`,
      width: `${percentage}%`
    };
  }
);
