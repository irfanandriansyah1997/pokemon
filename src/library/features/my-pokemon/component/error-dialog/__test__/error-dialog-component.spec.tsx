import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { DialogClose } from '@/library/component/dialog/style';
import ErrorDialog from '@/library/features/my-pokemon/component/error-dialog';

describe(`Testing Error Dialog Component`, () => {
  it(`Testing Snapshot Component`, () => {
    const onSpy = jest.fn((param) => param);
    const ui = shallow(<ErrorDialog on={onSpy} show />);

    expect(toJson(ui)).toMatchSnapshot();
  });

  it(`Simulate Close Dialog`, () => {
    const onSpy = jest.fn((param) => param);
    const ui = mount(<ErrorDialog on={onSpy} show />);

    ui.find(DialogClose).at(0).simulate(`click`);

    expect(onSpy).toHaveBeenCalledTimes(1);
    expect(onSpy.mock.results[0].value).toStrictEqual({ event: `on-close` });
  });
});
