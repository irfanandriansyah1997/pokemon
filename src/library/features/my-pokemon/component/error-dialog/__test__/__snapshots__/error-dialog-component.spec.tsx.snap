// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Error Dialog Component Testing Snapshot Component 1`] = `
<Dialog
  on={[MockFunction]}
  show={true}
  title="Failed"
>
  <Styled(div)>
    <Styled(p)>
      It ran away...
    </Styled(p)>
  </Styled(div)>
</Dialog>
`;
